backend/app/anti_recommendation_engine/anti_recommendation_engine.py

```diff
@@ -0,0 +1,74 @@
+from app.anti_recommenders.anti_recommender_generator import AntiRecommendationGenerator
+from app.models.record.record import Record
+from app.readers.all_source_reader import AllSourceReader
+
+
+class AntiRecommendationEngine:
+    """
+    The main AntiRecommendationEngine for the NerdSwipe backend.
+    It retrieves AntiRecommendations from the AntiRecommendationGenerator and possesses methods
+    which return tuples of Records that have the same titles as the generated AntiRecommendations or previous AntiRecommendations.
+    """
+
+    def __init__(self) -> None:
+        self.__record_store: dict[str, Record] = self.load_records()
+        self.__anti_recommendation_generator: AntiRecommendationGenerator = (
+            AntiRecommendationGenerator()
+        )
+        self.__stack: list[list[Record]] = []
+        self.__current_anti_recommendation_records: list[Record] = []
+
+    def load_records(self) -> dict[str, Record]:
+        """Load records from AllSourceReader and store them in __record_store."""
+
+        records = {}
+        all_source_reader = AllSourceReader()
+
+        for record in all_source_reader.read():
+            abstract_info = record.abstract_info
+            if abstract_info:
+                records[abstract_info.title] = record
+
+        return records
+
+    def generate_anti_recommendation_records(
```

Comment:
You could call this

`anti_recommend_related_records(self, to_record_key: str)`

`generate_initial_anti_recommendation_records(self) -> delegate to above...`